core/src/main/java/io/ddf/content/Schema.java

```diff
@@ -1,655 +1,193 @@
+  public void remove(int index) {
+    assert index >= 0 && index < columns.size();
+    Column column = columns.remove(index);
+    colNames.remove(column.getName());
+  }
 
-    for (int i = 0; i < mColumns.size(); i++) {
-      if (name.equalsIgnoreCase(mColumns.get(i).getName()))
-        return i;
+  public void remove(String colName) {
+    if (colNames.contains(colName)) {
+      columns.removeIf(a -> a.getName() == colName);
+      colNames.remove(colName);
     }
-
-    return -1;
   }
 
   @Override
   public String toString() {
-    Iterable<String> columnSpecs = Iterables.transform(getColumns(), new Function<Column, String>() {
-      @Override
-      public String apply(Column column) {
-        return column.getName() + " " + column.getType();
+    StringBuilder sb = new StringBuilder();
+    for(Column column : columns){
+      try {
+        sb.append(column.getName() + ":" + column.getType().getSimpleName() + "|");
+      }catch(Exception ex){
       }
-    });
-    return Joiner.on(",").join(columnSpecs);
-  }
 
-  /*
-   *
-   */
-  public void generateDummyCoding() throws NumberFormatException,
-      DDFException {
-    DummyCoding dummyCoding = new DummyCoding();
-    // initialize array xCols which is just 0, 1, 2 ..
-    dummyCoding.xCols = new int[this.getColumns().size()];
-    int i = 0;
-    while (i < dummyCoding.xCols.length) {
-      dummyCoding.xCols[i] = i;
-      i += 1;
     }
+    if(sb.length()>0)
+      sb.deleteCharAt(sb.length()-1);
+    return sb.toString();
+  }
 
-    List<Column> columns = this.getColumns();
-    Iterator<Column> columnIterator = columns.iterator();
-    int count = 0;
-    while (columnIterator.hasNext()) {
-      Column currentColumn = columnIterator.next();
-      int currentColumnIndex = this.getColumnIndex(currentColumn.getName());
-      HashMap<String, java.lang.Double> temp = new HashMap<String, java.lang.Double>();
-      // loop
-      if (currentColumn.getColumnClass() == ColumnClass.FACTOR) {
-        if (currentColumn.getOptionalFactor() != null && currentColumn.getOptionalFactor().getLevelMap() != null) {
-          Map<String, Integer> currentColumnFactor = currentColumn.getOptionalFactor().getLevelMap();
-          Iterator<String> valuesIterator = currentColumnFactor.keySet().iterator();
+  public static SchemaBuilder builder(){
+    return new SchemaBuilder() {
+      @Override
+      public Schema newSchema() {
+        return new Schema();
+      }
 
-          //TODO update this code
-          i = 0;
-          temp = new HashMap<String, java.lang.Double>();
-          while (valuesIterator.hasNext()) {
-            String columnValue = valuesIterator.next();
-            temp.put(columnValue, Double.parseDouble(i + ""));
-            i += 1;
-          }
-          dummyCoding.getMapping().put(currentColumnIndex, temp);
-          count += temp.size() - 1;
+      @Override
+      protected Class inferType(String typeName) throws SchemaException {
```

Comment:
Are these the only types we can support at the DDF level? Spark RDDs can contain other arbitrary types. This appears like a departure from our existing implementation.